dspace-spring-rest/src/main/java/org/dspace/app/rest/security/DSpacePermissionEvaluator.java

```diff
@@ -0,0 +1,44 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest.security;
+
+import java.io.Serializable;
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.access.PermissionEvaluator;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Component;
+
+@Component
+public class DSpacePermissionEvaluator implements PermissionEvaluator {
+
+    @Autowired
+    private List<RestPermissionEvaluatorPlugin> permissionEvaluatorPluginList;
```

Comment:
What are some examples of different plugins that might be in use at one time?  Is this to support the notion of "special groups" from earlier versions of DSpace, or is this a new concept?

Reply:
This is a new concept. For example, in DSpace 6 there are no resource policies attached to an EPerson object. This means the normal `AuthorizeService` cannot decide if authenticated EPerson A is allowed to view or update data of EPerson B. 

During one of the developer meetings, we decided to solve this problem only in the REST layer for now. Later (DSpace 8 or 9), this "authorize/permission plugin" system has to be pushed down to the "dspace-api" level.